src/server.c

```diff
@@ -3852,8 +3852,6 @@ static void readOOMScoreAdj(void) {
  * depending on current role.
  */
 int setOOMScoreAdj(int process_class) {
```

Comment:
It looks like this will cause CI to fail (in my daily CI)
mac: https://github.com/enjoy-binbin/redis/runs/4379281827?check_suite_focus=true#step:5:2246
freebsd: https://github.com/enjoy-binbin/redis/runs/4379282497?check_suite_focus=true#step:4:4682

Reply:
thanks for reporting.
the reason we removed it so that when this config is disabled (after being enabled), it'll remove the previously set adjustment.

i.e.
```
config set oom-score-adj-values 1 2 3
config set oom-score-adj yes
config set oom-score-adj no
```
i.e. the last line will undo the changes as if it was never enabled.

what i don't yet understand is why the `CONFIG sanity` fails on that.
IIRC what that test does it CONFIG GET, and then CONFIG SET with the value it got from GET.
and IIRC the config infra is suppose to catch these no-op things today and avoid calling the apply function.
can you please look into that?

regardless, i see another issue with that change.
let's say the user never enabled `oom-score-adj`, but as soon as he's setting `oom-score-adj-values` to a new value, we'll get here and try to apply the startup adjust (`oom_score_adj_base`).

Reply:
```
127.0.0.1:6379> config set oom-score-adj-values "0 200 800"
(error) ERR Config set failed - Failed to set current oom_score_adj. Check server logs.
```

> and IIRC the config infra is suppose to catch these no-op things today and avoid calling the apply function.

It looks like this failed, `createSpecialConfig` does not catch no-op things, those `special` commands are not checked that part

Reply:
ohh, right. special configs don't do that.
ok, the easy way out is to add these oom-score* configs to the `skip_configs` list.
the only reason this test used to pass so far is because it feeds back to redis the same value it got from CONFIG GET, and thus the oom-score-adj was always disabled, so even when it "modified" oom-score-adj-values, it didn't do anything.
can you please make a PR right away, so the daily CI doesn't fail?

the next thing would be to add explicit tests around these two configs:
1. to make sure that when disabling oom-score-adj, it reverts what it did when it was enabled.
2. make sure that setting either of these two configs when oom-score-adj is disabled, it will not try to modify `/proc` and fail on non-Linux systems.
3. consider improving the "special" oom-score-adj-values config to detect no-op changes and skip them.

we can leave that task of 3 tests and config improvement for @yoav-steinberg to handle when he's back.

in theory, if we did all 3 of the above (or even somehow just do 2, without breaking 1), there was no need to add them to the `skip_configs` list, but if we'll add specific sanity tests for them anyway, i think it's safe to skip them.

p.s. that whole `CONFIG sanity` is kinda useless now that everything in the config goes from a config framework... in the past each config had special hand written code, and this test (which is also relatively new) exposed some inconsistencies.

Reply:
So what I need to do now is to add `oom-score-adj` and `oom-score-adj-values` to skip_list?

I try it on my mac, the test also fails on `CONFIG SET rollback on apply error` ... i am working on it

Reply:
a quick fix, handled in #9881

Reply:
So I'll find a solution for 2 above where the user can manipulated `/proc` on their own and even if we write to `oom-score-adj-values` they don't need to worry redis will overwrite `/proc` as long as `oom-score-adj` is `no`.